jadx-gui/src/main/java/jadx/gui/utils/DesktopEntryUtils.java

```diff
@@ -0,0 +1,130 @@
+package jadx.gui.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jadx.core.export.TemplateFile;
+import jadx.core.utils.files.FileUtils;
+
+public class DesktopEntryUtils {
+	private static final Logger LOG = LoggerFactory.getLogger(DesktopEntryUtils.class);
+	private static final Map<Integer, String> SIZE_TO_LOGO_MAP = Map.of(
+			16, "jadx-logo-16px.png",
+			32, "jadx-logo-32px.png",
+			48, "jadx-logo-48px.png",
+			252, "jadx-logo.png",
+			256, "jadx-logo.png");
+
+	public static boolean createDesktopEntry() {
+		Path desktopTempFile = FileUtils.createTempFileNonPrefixed("jadx-gui.desktop");
+		Path iconTempFolder = FileUtils.createTempDir("logos");
+		LOG.debug("Creating desktop with temp files: {}, {}", desktopTempFile, iconTempFolder);
+		try {
+			return createDesktopEntry(desktopTempFile, iconTempFolder);
+		} finally {
+			try {
+				FileUtils.deleteFileIfExists(desktopTempFile);
+				FileUtils.deleteDirIfExists(iconTempFolder);
+			} catch (IOException e) {
+				LOG.error("Failed to clean up temp files", e);
+			}
+		}
+	}
+
+	private static boolean createDesktopEntry(Path desktopTempFile, Path iconTempFolder) {
+		String launchScriptPath = getLaunchScriptPath();
+		if (launchScriptPath == null) {
+			return false;
+		}
+		for (Map.Entry<Integer, String> entry : SIZE_TO_LOGO_MAP.entrySet()) {
+			Path path = iconTempFolder.resolve(entry.getKey() + ".png");
+			if (!writeLogoFile(entry.getValue(), path)) {
+				return false;
+			}
+			if (!installIcon(entry.getKey(), path)) {
+				return false;
+			}
+		}
+		if (!writeDesktopFile(launchScriptPath, desktopTempFile)) {
+			return false;
+		}
+		return installDesktopEntry(desktopTempFile);
+	}
+
+	private static boolean installDesktopEntry(Path desktopTempFile) {
+		try {
+			ProcessBuilder desktopFileInstallCommand = new ProcessBuilder("xdg-desktop-menu", "install", desktopTempFile.toString());
```

Comment:
```suggestion
			ProcessBuilder desktopFileInstallCommand = new ProcessBuilder("/usr/bin/xdg-desktop-menu", "install", desktopTempFile.toString());
```
Github code scanner suggest to use full path to execute binary, is `/usr/bin/` common for all distros?

Reply:
Although common, it is not guaranteed  
I added another function named `findExecutablePath` which will check `$PATH` for the executable and use the first valid match, which as far as I know is how you are supposed to find executables whose path is not known